# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/TSFTextStore.h
# Commit: 55825a1bc9fd
# Full Hash: 55825a1bc9fd5ca8e1568dd1bee6325ccd6127ed
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-30 09:35:36
# Regressor Bug: 1774317
# File Overlap Count: 2
# Description:
#   Bug 1774317 - part 3: Fix forgotten member initializing of `TSFTextStore` r=m_kato
#   
#   The crash tracked in bug 1792767 is caused by uninitialized member of
#   `TSFTextStore`.  Therefore, it runs the cleaning up the TSF objects at
#   setting focus to new `TSFTextStore`, but MS-IME didn't grab the necessary
# ==============================================================================

diff -r 1808f59fe299 -r 55825a1bc9fd widget/windows/TSFTextStore.h
--- a/widget/windows/TSFTextStore.h	Thu Sep 29 14:06:03 2022 +0000
+++ b/widget/windows/TSFTextStore.h	Thu Sep 29 14:06:03 2022 +0000
@@ -1035,7 +1035,7 @@
     // Count of the supported attributes
     NUM_OF_SUPPORTED_ATTRS
   };
-  bool mRequestedAttrs[NUM_OF_SUPPORTED_ATTRS]{false};
+  bool mRequestedAttrs[NUM_OF_SUPPORTED_ATTRS] = {false};
 
   int32_t GetRequestedAttrIndex(const TS_ATTRID& aAttrID);
   TS_ATTRID GetAttrID(int32_t aIndex);
@@ -1056,7 +1056,7 @@
   bool mWaitingQueryLayout = false;
   // During the document is locked, we shouldn't destroy the instance.
   // If this is true, the instance will be destroyed after unlocked.
-  bool mPendingDestroy;
+  bool mPendingDestroy = false;
   // If this is false, MaybeFlushPendingNotifications() will clear the
   // mContentForTSF.
   bool mDeferClearingContentForTSF = false;
